CRM/PivotReport/DataActivity.php

```diff
@@ -0,0 +1,374 @@
+<?php
+
+/**
+ * Provides a functionality to prepare entity data for Pivot Table.
+ */
+class CRM_PivotReport_DataActivity extends CRM_PivotReport_AbstractData {
+
+  public function __construct($name = NULL) {
+    $this->name = 'Activity';
+  }
+
+  /**
+   * Rebuilds pivot report cache including header and data.
+   *
+   * @param \CRM_PivotCache_AbstractGroup $cacheGroup
+   * @param array $params
+   *
+   * @return array
+   */
+  public function rebuildCache($cacheGroup, array $params) {
+    $this->fields = $this->getFields();
+    $this->emptyRow = $this->getEmptyRow();
```

Comment:
You can use the same trick of caching the data in `getFields` for this method and then always use only `getEmptyRow()` and avoid accessing the property directly